dataverse/orgsvc/C#/RemoveUnrefrencedQueues/README.md

```diff
@@ -0,0 +1,35 @@
+# Sample: Create a basic plug-in
+
+This sample shows how to write a simple plug-in that updates the activity parties on an email removing any related item's created by ARC that are no longer party to the email.  This assumes that the org has enabled ARC's Multi-Related for Email feature.
+
+## How to run this sample
+
+1. Download or clone the [Samples](https://github.com/Microsoft/PowerApps-Samples) repo so that you have a local copy. This sample is located under PowerApps-Samples-master\cds\orgsvc\C#\RemoveUnrefrencedQueues.
+2. Open the sample solution in Visual Studio, navigate to the project's properties, and verify the assembly will be signed during the build. Press F6 to build the sample's assembly (RemoveUnrefrencedQueues.dll).
+3. Run the Plug-in Registration tool and register the sample's assembly in the D365 server's sandbox and database. When registering a step, specify the Create message, email table, and syncronous mode on Post Create stage.
```

Comment:
**So far, I have only glanced at the code.** But performance for any synchronous step is important. When I look at the code, I'll expect to find some guards to ensure it won't time out or have a bad experience for app users.

Does this have to be synchronous?

I'll read more and try to find out...

Reply:
I know this PR will be abandoned but did not want to loose the work you had here.

It does for this purpose but you wont read anything to explain why.  I'll add that. (TLDR they don't want end users to ever see this email on the removed related entities timelines). 

There are two main use cases:

1. C2 has a case that was opened by sending an email, all the back and forth there is captured on the case's timeline.  C2 forwards an email from that chain to the agent's boss.  C1 does not want that forwarded email on the timeline.  However by default it will be because of SSS's correlation logic (that cant be customized)

2. C2 has sent an email to the support queue and engineering email queues.  Two cases are created by ARC (Automatic Record Creation).  C2 then replies from their sent mailbox (yes really they do this apparently) and removes the engineering queue.  C1 wants the reply that was only sent to the support queue to only show up on the Support Case's Timeline (not show up on the Engineering Case's timeline)